R/functions_netCDF.R

```diff
@@ -188,36 +427,79 @@ create_netCDF_from_raster_with_variables <- function(x, time_bounds,
     vals = rbind(yvals + grid_halfres[2], yvals - grid_halfres[2]),
     start = c(1, 1), count = c(2L, var_chunksizes[2])))
 
-  if (has_time_central) {
-    try(ncdf4::ncvar_put(nc, varid = "time_bnds",
-      vals = time_bounds,
-      start = c(1, 1), count = c(2L, 1L)))
+  if (has_T_timeAxis) {
+    if(has_time_central){
+      try(ncdf4::ncvar_put(nc, varid = "time_bnds",
+                           vals = time_bounds,
+                           start = c(1, 1), count = c(2L, 1L)))
+    } else {
+      try(ncdf4::ncvar_put(nc, varid = "time_bnds",
+                           vals = time_bounds,
+                           start = c(1, 1), count = c(2L, t_chunksize))) # beginning and end of each TP
+    }
   }
 
+  if(has_Z_verticalAxis) {
+    try(ncdf4::ncvar_put(nc, varid = "depth_bnds",
+                         vals = vert_bounds,
+                         start = c(1, 1), count = c(2L, z_chunksize))) # top and bottom of each soil layer
+  }
+
+  #--- add attributes -----------------------------------------------------------------
 
-  #--- add attributes
-  # add dimension attributes
+  # add dimension attributes --------------------------------
   ncdf4::ncatt_put(nc, "lon", "axis", "X")
   ncdf4::ncatt_put(nc, "lon", "bounds", "lon_bnds")
   ncdf4::ncatt_put(nc, "lat", "axis", "Y")
   ncdf4::ncatt_put(nc, "lat", "bounds", "lat_bnds")
 
-  if (has_time_central) {
+  if (has_T_timeAxis) {
     ncdf4::ncatt_put(nc, "time", "axis", "T")
     ncdf4::ncatt_put(nc, "time", "bounds", "time_bnds")
-    ncdf4::ncatt_put(nc, "time", "calendar", "gregorian")
   }
 
-  # add global attributes
+  if (has_Z_verticalAxis) {
+    ncdf4::ncatt_put(nc, "depth", "axis", "Z")
+    ncdf4::ncatt_put(nc, "depth", "bounds", "depth_bnds")
+  }
+
+  for (natt in ns_att_vert) {
+    ncdf4::ncatt_put(nc, varid = vert_names, attname = natt,
+                     attval = vertical_attributes[[ns_att_vert]][1])
+  }
+
+  # add variable attributes  ---------------------------------------------------
+  for (k in seq_len(nn)) {
+    for (natt in ns_att_vars) {
+      ncdf4::ncatt_put(nc, varid = var_names[k], attname = natt,
+                       attval = var_attributes[[natt]][k])
+    }
+  }
+
+  # add coordinate system attributes --------------------------------------------
+  if(is.null(grid)){
+    prj <- raster::crs(grid)
+  } else {
+    prj <- sp::CRS(crs)
+  }
+
+  if (!is.na(prj)) {
+    ncdf4::ncatt_put(nc, varid = "crs", attname = "proj4",
```

Comment:
I think we should change how we handle CRS to what CF-1.8 has: http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#grid-mappings-and-projections, e.g., see "Example 5.9. Latitude and longitude on the WGS 1984 datum", and provide our CRS string as attribute name `crs_wkt` instead (see http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#use-of-the-crs-well-known-text-format)

Reply:
added